app/tasksmodule/datachecks.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tasks to check data.
+
+These tasks don't modify data, they just double-check things that are already
+expected to be true.
+"""
+
+import datetime
+import time
+
+import django.http
+from google.appengine import runtime
+from google.appengine.api import datastore_errors
+from google.appengine.api import taskqueue
+
+import tasksmodule.base
+import model
+import utils
+import views.base
+
+
+class DatacheckException(Exception):
+    """An exception raised when a datacheck tasks finds a problem."""
+    pass
+
+
+class DatachecksBaseTask(tasksmodule.base.PerRepoTaskBaseView):
+
+    def get_params(self):
+        return views.base.read_params(
+            super(DatachecksBaseTask, self).get_params(),
+            self.request,
+            post_params={'cursor': utils.strip})
+
+    def schedule_task(self, repo, **kwargs):
+        name = '%s-%s-%s' % (repo, self.BASE_NAME, int(time.time()*1000))
+        path = self.build_absolute_path('/%s/tasks/%s' % (repo, self.TASK_PATH))
+        cursor = kwargs.get('cursor', '')
+        # TODO(nworden): figure out why setting task_retry_limit isn't working
+        retry_options = taskqueue.taskqueue.TaskRetryOptions(task_retry_limit=1)
+        taskqueue.add(name=name, method='POST', url=path,
+                      queue_name='datachecks', retry_options=retry_options,
+                      params={'cursor': cursor})
+
+    def alert(self, msg):
+        """Alerts developers of an error.
+
+        For now, we just raise an exception. It will show up in Stackdriver.
+        """
+        raise DatacheckException(msg)
+
+
+class PersonDataValidityCheckTask(DatachecksBaseTask):
+    """Checks that person records have valid data."""
+
+    BASE_NAME = 'person_data_validity_check'
+    TASK_PATH = 'check_person_data_validity'
+
+    def _check_person(self, person):
+        if not person.entry_date:
+            self.alert('A person record is missing an entry_date value.')
```

Comment:
just checking: do we want to add troubleshooting info to the alert message? (like the record id) otherwise receiving the alert would require running the same task that the check performed essentially.

Reply:
Good idea, done.